src/api/AppActivity.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * @flow
+ * @file Helper for the box App Activity API
+ * @author Box
+ */
+import MarkerBasedAPI from './MarkerBasedAPI';
+import { ERROR_CODE_DELETE_APP_ACTIVITY, HTTP_STATUS_CODE_NOT_FOUND } from '../constants';
+import { APP_ACTIVITY_FIELDS_TO_FETCH } from '../utils/fields';
+
+class AppActivity extends MarkerBasedAPI {
+    /** @property {TasksAPI} - Placeholder permissions object to determine if app activity can be deleted */
+    permissions: BoxItemPermission = {};
+
+    /**
+     * Map an entry from the AppActivity API to an AppActivityItem.
+     * occurred_at -> created_at
+     *
+     * @param {Object} item - A single entry in the AppActivity API entiries list
+     *
+     * @return {AppActivityItem}
+     */
+    mapAppActivityItem = (item: Object): AppActivityItem => {
+        const { occurred_at, ...rest } = item;
+        const { can_delete } = this.permissions;
+        return {
+            created_at: occurred_at,
+            permissions: {
+                can_delete,
+            },
+            ...rest,
+        };
+    };
+
+    /**
+     * API URL for getting App Activity on a file
+     *
+     * @return {string} Url for all app activity on a file
+     */
+    getUrl(): string {
+        return `${this.getBaseApiUrl()}/app_activities`;
+    }
+
+    /**
+     * API URL for deleting app activity from a file
+     *
+     * @param id - ID of an app activity item
+     * @return {string} - URL to delete app activity
+     */
+    getDeleteUrl(id: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getUrl()}/${id}`;
+    }
+
+    /**
+     * Generic success handler
+     *
+     * @param {AppActivityItems} data - the response data
+     */
+    successHandler = ({ entries = [] }: AppActivityItems): void => {
+        if (this.isDestroyed() || typeof this.successCallback !== 'function') {
+            return;
+        }
+
+        const activityEntries = entries.map(this.mapAppActivityItem);
+        this.successCallback({ entries: activityEntries, total_count: activityEntries.length });
+    };
+
+    /**
+     * Generic error handler
+     *
+     * @param {AxiosError} error - the error response
+     */
+    errorHandler = (error: $AxiosError<any>): void => {
+        if (this.isDestroyed() && typeof this.errorCallback !== 'function') {
+            return;
+        }
+
+        this.permissions = {};
```

Comment:
Why do we need to wipe the permissions? It looks like it gets reset on every call to `getAppActivity`.

Reply:
In the case of an error, I thought a reset would be safe, but now I'm not convinced it makes a difference (as you've pointed out)